idb/postgres/internal/writer/writer.go

```diff
@@ -61,16 +60,18 @@ var statements = map[string]string{
 		localstate = EXCLUDED.localstate, deleted = FALSE`,
 	deleteAccountStmtName: `INSERT INTO account
 		(addr, microalgos, rewardsbase, rewards_total, deleted, created_at, closed_at,
-			account_data)
-		VALUES($1, 0, 0, 0, TRUE, $2, $2, 'null'::jsonb) ON CONFLICT (addr) DO UPDATE SET
+		 keytype, account_data)
+		VALUES($1, 0, 0, 0, TRUE, $2, $2, $3, 'null'::jsonb) ON CONFLICT (addr) DO UPDATE SET
 		microalgos = EXCLUDED.microalgos, rewardsbase = EXCLUDED.rewardsbase,
 		rewards_total = EXCLUDED.rewards_total, deleted = TRUE,
-		closed_at = EXCLUDED.closed_at, account_data = EXCLUDED.account_data`,
+		closed_at = EXCLUDED.closed_at, keytype = EXCLUDED.keytype,
```

Comment:
Right, this sql query will not clear it.

Reply:
There are some weird edge cases. For example if you tried to rekey an account that was being deleted, you would skip clearing the sig type, unless it already had an auth-addr in which case it would still need to be cleared.

I don't see the code that checks for auth-addr and conditionally clears it.

Reply:
I think there is no problem. Here is where clearing happens. https://github.com/algorand/indexer/pull/783/files#diff-940e0e8ce1ffe2b07a3c71d6bb32f7b74b918d0c9d4661d9b3fb339dc5b11b18R383

Reply:
There are 3 cases:
1) `RekeyTo` is set, we must clear the sigtype field.
2) `RekeyTo` is not set, sigtype is not set, we must set the sigtype field.
3) `RekeyTo` is not set, sigtype is set, setting the sigtype field is not necessary.

The 3 states can be used to handle each of these cases individually, but currently the 3rd is not triggering any different behavior.